test/realm_test.dart

```diff
@@ -1756,6 +1755,125 @@ Future<void> main([List<String>? args]) async {
       });
     }
   }
+  test('Realm.refresh no changes', () async {
+    final realm = getRealm(Configuration.local([Person.schema]));
+    final result = realm.refresh();
+    expect(result, false);
+  });
+
+  test('Realm.refresh into sync transaction', () async {
```

Comment:
what does it mean `into sync transaction`. This test name is not clear. Also it is named `Realm.refresh()` but executes `refreshAsync`
Should we rename to 
```suggestion
  test('Realm.refreshAsync() sync transaction', () async {
```